docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/rs-configuration.md

```diff
@@ -182,16 +182,259 @@ The capacity usage on a pool can be monitored using [exporter metrics](../replic
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  nodeAffinityTopologyLabel: |
+    zone: us-west-1
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+Apply the labels to the nodes using the below command:
+
+**Command**
+```text
+kubectl mayastor label node worker-node-1 zone=us-west-1
+kubectl mayastor label node worker-node-2 zone=eu-east-1
+kubectl mayastor label node worker-node-3 zone=us-west-1
+ ```
+
+**Command (Get nodes)**
+ ```text
+kubectl mayastor get nodes -n openebs --show-labels
+ID             GRPC ENDPOINT        STATUS  LABELS
+worker-node-1  65.108.91.181:10124  Online  zone=eu-west-1
+worker-node-3  65.21.4.103:10124    Online  zone=eu-east-1
+worker-node-3  37.27.13.10:10124    Online  zone=us-west-1
+```
+
+### "NodeHasTopologyKey"
+
+The parameter 'NodeHasTopologyKey' will allow the placement of replicas on the node that has label keys that are identical to the keys specified in the storage class.
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  nodeHasTopologykey: |
+    rack
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+Apply the labels on the node using the below command:
+
+**Command**
+```text
+ # kubectl mayastor label node worker-node-1 rack=1
+ # kubectl mayastor label node worker-node-2 rack=2
+ # kubectl mayastor label node worker-node-3 rack=2
+ 
+ # kubectl mayastor get nodes -n openebs --show-labels
+ ID             GRPC ENDPOINT        STATUS  LABELS
+ worker-node-1  65.108.91.181:10124  Online  rack=1
+ worker-node-3  65.21.4.103:10124    Online  rack=2
+ worker-node-3  37.27.13.10:10124    Online  rack=2
+```
+
+In this case, the replica of the volume will be placed on any of the two nodes i.e. 
+- `worker-node-1` and `worker-node-2` or 
+- `worker-node-1` and `worker-node-3` or
+- `worker-node-2` and `worker-node-3`
+as the storage class has `rack` as the value for `nodeHasTopoogyKey` that matches the label key of the node.
+
+### "NodeSpreadTopologyKey"
+
+The parameter 'NodeSpreadTopologyKey' will allow the placement of replicas on the node that has label keys that are identical to the keys specified in the storage class but have different values.
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  nodeSpreadTopologyKey: |
+    zone
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+Apply the labels to the nodes using the below command:
+
+**Command**
+```text
+kubectl mayastor label node worker-node-1 zone=us-west-1
```

Comment:
@sinhaashish Why are we not using 'kubectl label node' instead of 'kubectl mayastor label node'?
cc @tiagolobocastro @Abhinandan-Purkait your thoughts?

Reply:
This is labeling the control-plane node structure and not the k8s node. Currently we don't have any way to propagate the k8s node label to mayastor node that's why this is being done this way, iiuc.

Reply:
Hmm, thanks for the explanation.
We may be missing a point here, I think users may prefer K8s labels for nodes and diskpool CR.

Reply:
Yes that's true, but we would need some sort of operator to sync them with control-plane. We were thinking of adding some script that could do that, iirc. @sinhaashish can confirm.

Reply:
yes i can recall now. to sync the labels of k8s notes to mayastor nodes, operator is needed. So for the first cut we went with labelling the mayastor nodes using the `kubectl mayastor` command only.